include/simdjson/generic/ondemand/parser.h

```diff
@@ -230,15 +230,14 @@ class parser {
   /** The maximum capacity of this parser (the largest document it is allowed to process). */
   simdjson_inline size_t max_capacity() const noexcept;
   simdjson_inline void set_max_capacity(size_t max_capacity) noexcept;
-  /** The maximum depth of this parser (the most deeply nested objects and arrays it can process). */
+  /** Deprecated method. */
   simdjson_inline size_t max_depth() const noexcept;
-
   /**
    * Ensure this parser has enough memory to process JSON documents up to `capacity` bytes in length
    * and `max_depth` depth.
    *
    * @param capacity The new capacity.
-   * @param max_depth The new max_depth. Defaults to DEFAULT_MAX_DEPTH.
+   * @param max_depth The new max_depth. Defaults to DEFAULT_MAX_DEPTH. (This parameter has no effect.)
```

Comment:
Applications using this library may run into bugs if they assume it really has no effect.

It has an effect; the dom parser implementation will allocate more memory if a higher max_depth is used. (9x the depth for open_container+is_array?)

Maybe choosing a smaller depth would be more appropriate, or 0 if possible

```
inline simdjson_warn_unused error_code dom_parser_implementation::set_max_depth(size_t max_depth) noexcept {
  // Stage 2 stacks
  open_containers.reset(new (std::nothrow) open_container[max_depth]);
  is_array.reset(new (std::nothrow) bool[max_depth]);
  if (!is_array || !open_containers) { _max_depth = 0; return MEMALLOC; }

  _max_depth = max_depth;
  return SUCCESS;
}
```